proposals/0005-jwt-rbac.md

```diff
@@ -0,0 +1,161 @@
+```json
+  {
+   		"iss": "https://server.example.com",
+   		"sub": "24400320",
+   		"aud": "s6BhdRkqt3",
+   		"nonce": "n-0S6_WzA2Mj",
+   		"exp": 1311281970,
+   		"iat": 1311280970,
+   		"auth_time": 1311280969,
+	}
+```
+Please, note that OpenID Connect specification also defines a few other standard claims that can be used to identity the subject and improve the security of a token. The snippet above is just the bare minimum necessary to represent a subject with basic information about how tokens should be validated. Also note that the ID Token is not meant to be used as a bearer/access token to gain access to resources protected by a resource server or service. Here we propose a bearer/access token format, based on some claims from ID Token.
+
+In addition to these standard claims, we also recommend a **preferred_username** claim to carry the username. For instance:
+```json
+  {
+   		"iss": "https://server.example.com",
+   		"sub": "24400320",
+      "preferred_username": "jdoe",
+   		"aud": "s6BhdRkqt3",
+   		"nonce": "n-0S6_WzA2Mj",
+   		"exp": 1311281970,
+   		"iat": 1311280970,
+   		"auth_time": 1311280969,
+	  }
+```
+
+From an authorization perspective, we recommend two specific claims to carry role information about a subject.
+```json
+	{
+   		"iss": "https://server.example.com",
+   		"sub": "24400320",
+     "preferred_username": "jdoe",
+   		"aud": "s6BhdRkqt3",
+   		"nonce": "n-0S6_WzA2Mj",
+   		"exp": 1311281970,
+   		"iat": 1311280970,
+   		"auth_time": 1311280969,
+      "realm_access": {
+          "roles": ["role-in-realm", "user", "manager"]
+      },
+      "resource_access": {
+          "my-service": {
+            "roles": [
+              "role-in-my-service"
+            ]
+          }
+        },
+      }
+  }
+```
+The **realm_access** claim can be used to define which roles were granted to a subject at the realm or security domain level. On the other hand, the **resource_access** claim allows to define a different set of roles which are associated with a specific resource server or service.
+
+It is recommend that signature portion of the token should be signed using the RS256 algorithm using the RSA private key associated with the realm so that clients and services may verify the signature using the RSA realm public key. This corresponds to a JWT header with the following **alg** and **typ** claims:
+```
+{
+  "alg": "RS256",
+  "typ": "JWT",
+  "kid": "..."
+}
+```
+
+## Impact on existing code (if applicable)
+
+Existing services that desire to implement JWT RBAC must configure the container to support integration with an authorization implementation that is capable of parsing the JWT bearer token to validate the token and populate a security context with the caller and granted roles for use in ```getCallerPrincipal()/getUserPrincipal(), isCallerInRole(String)/isUserInRole(String)``` type of container methods.
```

Comment:
@starksm64 okay, great ;) At a minimum it may be a good idea to look at the terminology that JSR 375 is using and where applicable use the same terms for micro profile.